### PR TITLE
Add event-based handler templates documentation

### LINKED ISSUE #2541: Surface better place to discuss Golang templates + handlers
### Description

Our handlers (e.g., the Slack handler & others) often make use of Golang templates. While we can link to the Golang docs to generally discuss templating as a concept, there doesn't seem to be a way that we can discuss the templated attributes in a concrete way. The path I've taken personally is to use the attributes in the filter docs to suss out what template attributes are available, and I'd hazard a guess that our users are left to do the same. 

Ideally, we'd have a doc that discusses these attributes as they pertain to handlers so that they're easily surfaced for our users.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [x] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### LINKED ISSUE #2724: Add page to document templating
### Category

This is an issue with:

- [ ] errata (error in doc content)
- [x] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
We probably need to link to the new doc from the handler and event references, as well as READMEs for handler plugins.

### Expected behavior
The docs include a page that describes templating from events and how handlers use it. The new doc covers basic template usage and what’s available from the event (rather than linking to the event reference).

### Current behavior
The docs do not describe templating from events.

### Corrections needed
Add a doc page devoted to templating.

### Additional context
https://sensu.slack.com/archives/C3MHLUP46/p1599825252309900
https://sensucommunity.slack.com/archives/C68LV5M9U/p1599670740009500

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds a page in Observability Pipeline > Process to document handler templates

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2724
Closes https://github.com/sensu/sensu-docs/issues/2541

### CODE REVIEW

**Review by nixwiz — APPROVED** [review]
LGTM!

**Comment by hillaryfraley** [comment]
Maybe some of the instructions about how to use the template toolkit should be in the readme instead?

**Comment by nixwiz** [comment]
> Maybe some of the instructions about how to use the template toolkit should be in the readme instead?

Definitely, I think @jspaleta was more concerned with getting the prototype code out.  I'm sure the README will be made much better.

**Comment by hillaryfraley** [comment]
Oh yes, I was just thinking whether I should move some of this to a commit against the readme. I'll defer to @jspaleta on whether he wants to use any of it for the readme. Thank you @nixwiz !

**Comment by hillaryfraley** [comment]
I still need to confirm how the metadata and system attributes work -- can't get the command working locally.

Also need to confirm the commands using sensuctl to pull events from API.

File: content/sensu-go/6.0/observability-pipeline/observe-process/handler-templates.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+| `.Entity.System` | `.Entity.System.Arch` | `.Entity.System.ARMVersion` |
+| `.Entity.System.CloudProvider` | `.Entity.System.Hostname` | `.Entity.System.LibcType` |
+| `.Entity.System.Network` | `.Entity.System.OS` | `.Entity.System.Platform` |
+| `.Entity.System.PlatformFamily` | `.Entity.System.PlatformVersion` | `.Entity.System.Processes` |
+| `.Entity.System.VMRole` | `.Entity.System.VMSystem` | `.Entity.User` |
+| `.Metrics.Handlers` | `.Metrics.Points` | |
+
+## Template toolkit command
+
+The [Sensu template toolkit command][8] is a Sensu handler plugin you can use to print a list of available event attributes in handler template dot notation syntax and validate your handler template output.
```
Comment (nixwiz) [inline content/sensu-go/6.0/observability-pipeline/observe-process/handler-templates.md]: It's actually a sensuctl command plugin, not a handler plugin.

Comment (hillaryfraley) [inline content/sensu-go/6.0/observability-pipeline/observe-process/handler-templates.md]:
```suggestion
The [Sensu template toolkit command][8] is a sensuctl command plugin you can use to print a list of available event attributes in handler template dot notation syntax and validate your handler template output.
```

**Pull request merged**